hibernate-rx-core/src/test/java/org/hibernate/rx/TestConfiguration.java

```diff
@@ -0,0 +1,40 @@
+package org.hibernate.rx;
+
+import org.testcontainers.containers.MySQLContainer;
+import org.testcontainers.containers.PostgreSQLContainer;
+
+public class TestConfiguration {
+  
+  private static final boolean USE_DOCKER = Boolean.getBoolean("docker");
+  
+  private static final MySQLContainer<?> mysql = new MySQLContainer<>()
+        .withUsername("hibernate-rx")
+        .withPassword("hibernate-rx")
+        .withDatabaseName("hibernate-rx")
+        .withReuse(true);
+  
+  private static final PostgreSQLContainer<?> postgresql = new PostgreSQLContainer<>()
+        .withUsername("hibernate-rx")
+        .withPassword("hibernate-rx")
+        .withDatabaseName("hibernate-rx")
+        .withReuse(true);
+  
+  public static String getPostgreSQLURL() {
+    if (USE_DOCKER) {
+      postgresql.start();
+      return postgresql.getJdbcUrl() +  "&user=" + postgresql.getUsername() + "&password=" + postgresql.getPassword();
+    } else {
+      return "jdbc:postgresql://localhost:5432/hibernate-rx?user=hibernate-rx&password=hibernate-rx";
```

Comment:
In this PR conceptually Docker==Testcontainers, so perhaps if I renamed things to `USE_TESTCONTAINERS`/`-Ptestcontainers` that would be more correct?

The only real difference in the URLs will be the port -- Testcontainers will expose the DB on a random (free) port so you don't need to worry about port conflicts. The other elements are effectively hard-coded, but are declared earlier in the file (e.g. `withUsername("hibernate-rx")`), so I thought it would be better to not re-hardcode that string in multiple places.

Reply:
> In this PR conceptually Docker==Testcontainers, so perhaps if I renamed things to USE_TESTCONTAINERS/-Ptestcontainers that would be more correct?

I think docker is fine, it doesn't matter if it's testcontainers or not. Underneath is docker.